ui/admin/app/components/agent/Agent.tsx

```diff
@@ -41,46 +39,31 @@ export function Agent({ className, currentThreadId, onRefresh }: AgentProps) {
 		useAgent();
 
 	const [agentUpdates, setAgentUpdates] = useState(agent);
-	const [loadingAgentId, setLoadingAgentId] = useState("");
 
 	useEffect(() => {
-		setAgentUpdates((prev) => {
-			if (agent.id === prev.id) {
-				return {
-					...prev,
-					aliasAssigned: agent.aliasAssigned,
-				};
-			}
-
-			return agent;
-		});
-	}, [agent]);
-
-	const getLoadingAgent = useSWR(
-		AgentService.getAgentById.key(loadingAgentId),
-		({ agentId }) => AgentService.getAgentById(agentId),
-		{
-			revalidateOnFocus: false,
-			refreshInterval: 2000,
+		if (agent.aliasAssigned === undefined && agent.alias) {
```

Comment:
I refactored the logic a bit so that

1. When agent.aliasAssigned is not set and agent.alias is set, we should poll agent until it is being set so we know the final state.
2. Once UI makes a update for alias, set aliasAssigned to null and have it repoll agent until it is set.